Logging/LogHandler/FileLogHandler.cs

```diff
@@ -97,6 +158,56 @@ protected override void DisposeSelf()
                 );
 
             m_Writer.WriteLine($"{timestamp}{logLevel}{context}{message}");
+
+            if (m_Rotate && m_Writer.BaseStream.Length > RotateFileSizeLimit)
+            {
+                this.RotateFiles();
+            }
+        }
+
+        private void RotateFiles()
+        {
+            m_Writer.Close();
+            m_Writer = null;
+
+            // Determine how many rotated log files already exist
+            int index = 1;
+            string rotatedFilePath = GetRotatedFilePath(index);
+            while (File.Exists(rotatedFilePath))
+            {
+                rotatedFilePath = GetRotatedFilePath(++index);
+            }
+            --index;
+
+            // Delete any files in excess of the file count limit
+            while (index >= RotateFileCountLimit)
+            {
+                File.Delete(GetRotatedFilePath(index));
+                --index;
+            }
+
+            // Rotate remaining files
+            while (index >= 0)
+            {
+                File.Move(GetRotatedFilePath(index), GetRotatedFilePath(index + 1));
+                --index;
+            }
+
+            CreateWriter();
+        }
+
+        private string GetRotatedFilePath(int index)
+        {
+            if (index == 0)
+            {
+                return m_Path;
+            }
+
+            string directory = Path.GetDirectoryName(m_Path);
+            string fileName = Path.GetFileNameWithoutExtension(m_Path);
+            string extension = Path.GetExtension(m_Path);
+            
+            return Path.Combine(directory, $"{fileName}.{index}{extension}");
```

Comment:
Minor since this wouldn't get called often, but we can cache these strings as readonly in the constructor.